### ISSUE TITLE
[🐛] Adding custom notification channel ID to firebase.json results in FIR_AUTH_ERROR

### ISSUE THREAD

**Issue by royderksroute42** [invertase/react-native-firebase#4765]
🔥 

## Issue
```
2021-01-12 12:34:10.868 30922-31179/redactedbundleid E/ReactNativeJS: { [Error: [messaging/unknown] FIS_AUTH_ERROR]
      line: 26183,
      column: 28,
      sourceURL: 'http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=redactedbundleid&modulesOnly=false&runModule=true' }
```

The above is the _logcat_ error.
This crash only occurs on the first call to firebase().getToken(). Restarting the app afterwards 'fixes' it (the custom channel works, theres no error and the popup shows when the app is in the background).
The crash started happening after i specified a custom notification channel name by adding a firebase.json to my project root.
I ripped the firebase.json below straight from the docs, and the notification channel works flawlessly.
Looking at other issues, i learned to look for an error in logcat, and tried updating the google-services.json from firebase but to no avail.

Please note that the snippets below are the only thing we do with firebase; cloud messaging. And it works perfectly after the first crash OR when omitting the firebase.json file. Messages arrive with sound but without a popup when the app is on the background.
I had to introduce a custom android notification channel due to the fact that the Firebase misc channel had "sound" instead of "sound and popup" settings in the apps notifications channels info.

Creating channel using react-native-push-notification
```
export const CHANNEL_ID = 'High-Priority';
PushNotification.configure({
  onRegister(token) {
    console.log('TOKEN:', token);
  },
  onNotification(notification) {
    notification.finish(PushNotificationIOS.FetchResult.NoData);
  },
  requestPermissions: false,
});

PushNotification.createChannel(
      {
        channelId: CHANNEL_ID, // (required)
        channelName: CHANNEL_ID, // (required)
        channelDescription: CHANNEL_ID, // (optional) default: undefined.
        soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function
        importance: 4, // (optional) default: 4. Int value of the Android notification importance
        vibrate: true, // (optional) default: true. Creates the default vibration patten if true.
      }
    );
```

Relevant parts of a class that houses the functions.
```
 /** Token */

  /**
   * Returns the current token or null if not permitted
   */
  currentPushToken = (callback: (token: Optional<string>) => void) =>
    this.hasPermittedFCM(permitted =>
      permitted
        ? this.currentTokenGeneration(callback)
        : this.requestPermission(reqPermission =>
            reqPermission
              ? this.currentTokenGeneration(callback)
              : callback(null)
          )
    );

  /**
   * Generate the current token on demand
   */
  currentTokenGeneration = (callback: (__: string | null) => void) => {
    firebase() // <=== crash is here
      .getToken()
      .then(fcmToken => {
        if (fcmToken) {
          callback(fcmToken);
          firebase().onTokenRefresh(t => {
            console.warn('FB TOKEN: ', +t);
            this._onTokenGenerated(t);
          });
        } else {
          callback(null);
        }
      })
      .catch(e => {
        console.error(e);
      });
  };

  /**
   * Checks to see if Push Messages have been permitted
   */
  hasPermittedFCM = (callback: (hasPermitted: boolean) => void) => {
    firebase()
      .hasPermission()
      .then(enabled => {
        // console.warn(`FCM enabled: ${enabled}`);
        callback(enabled === 1);
      })
      .catch(e => {
        console.error(e);
      });
  };

  /**
   * Request permission to use Push Messages
   */
  requestPermission = (callback: (result: boolean) => void) => {
    firebase()
      .requestPermission()
      .then(() => {
        // User has authorised
        callback(true);
      })
      .catch(error => {
        // User has rejected permissions
        console.log(error);
        callback(false);
      });
  };
```

Describe your issue here

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
{
  "dependencies": {
    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.2.0",
    "@codler/react-native-keyboard-aware-scroll-view": "^1.0.1",
    "@react-native-community/datetimepicker": "^3.0.8",
    "@react-native-community/masked-view": "^0.1.10",
    "@react-native-community/push-notification-ios": "^1.8.0",
    "@react-native-firebase/app": "10.4.0",
    "@react-native-firebase/messaging": "10.4.1",
    "@react-native-picker/picker": "1.9.4",
    "@sentry/react-native": "^2.1.0",
    "@terrylinla/react-native-sketch-canvas": "^0.8.0",
    "@types/remotedev-serialize": "^1.0.0",
    "@voximplant/react-native-foreground-service": "^2.0.0",
    "appcenter": "4.0.0",
    "appcenter-analytics": "4.0.0",
    "appcenter-crashes": "4.0.0",
    "babel-plugin-transform-remove-console": "^6.9.4",
    "base64-js": "^1.3.0",
    "color": "^3.0.0",
    "core-js": "^2.6.5",
    "haversine": "^1.1.1",
    "hoist-non-react-statics": "^3.3.0",
    "immutable": "4.0.0-rc.12",
    "isomorphic-fetch": "^2.2.1",
    "lodash": "^4.17.10",
    "metro-config": "^0.56.0",
    "moment": "^2.22.2",
    "re-reselect": "^2.1.0",
    "react": "17.0.1",
    "react-dom": "^16.7.0",
    "react-native": "0.64.0-rc.1",
    "react-native-android-open-settings": "^1.2.0",
    "react-native-background-fetch": "^3.1.0",
    "react-native-background-geolocation-android": "https://github.com/bundle/react-native-background-geolocation-android.git#3.9.3",
    "react-native-barcode-mask": "^1.1.0",
    "react-native-camera": "^3.7.1",
    "react-native-collapsible": "^1.0.0",
    "react-native-device-info": "^0.22.1",
    "react-native-dialogs": "^1.0.2",
    "react-native-flipper": "^0.69.0",
    "react-native-fs": "2.16.6",
    "react-native-gesture-handler": "^1.9.0",
    "react-native-grid-list": "^1.1.0",
    "react-native-htmlview": "^0.15.0",
    "react-native-i18n": "https://github.com/bundle/react-native-i18n/tarball/f806f771f989b5d79f2ed00df9382c4ccce1d5b9",
    "react-native-image-crop-picker": "https://github.com/bundle/react-native-image-crop-picker/tarball/v0.21.5",
    "react-native-ios-picker": "^1.0.0",
    "react-native-keep-awake": "^4.0.0",
    "react-native-modal": "^6.5.0",
    "react-native-modal-datetime-picker": "^9.1.0",
    "react-native-navbar": "^2.1.0",
    "react-native-open-maps": "^0.3.1",
    "react-native-orientation": "^3.1.3",
    "react-native-paper": "3.0.0-alpha.5",
    "react-native-pdf": "^5.0.9",
    "react-native-permissions": "^1.1.1",
    "react-native-picker-select": "^8.0.4",
    "react-native-pod": "^1.12.1",
    "react-native-push-notification": "^6.1.3",
    "react-native-reanimated": "^1.13.2",
    "react-native-safe-area-context": "^3.1.9",
    "react-native-screens": "^2.16.1",
    "react-native-sensors": "^7.0.2",
    "react-native-sound": "^0.11.0",
    "react-native-status-bar-height": "^2.0.0",
    "react-native-svg": "^12.1.0",
    "react-native-vector-icons": "^6.1.0",
    "react-native-wakeful": "^0.1.0",
    "react-native-webview": "^10.9.0",
    "react-navigation": "^4.4.3",
    "react-navigation-drawer": "^2.6.0",
    "react-navigation-hooks": "^1.1.0",
    "react-navigation-stack": "^2.10.2",
    "react-navigation-tabs": "^2.10.1",
    "react-redux": "^7.1.0",
    "redux": "^4.0.0",
    "redux-devtools-extension": "^2.13.5",
    "redux-flipper": "^1.4.2",
    "redux-form": "^8.2.4",
    "redux-observable": "^1.1.0",
    "redux-on-state-change": "^1.0.0",
    "redux-persist": "^5.10.0",
    "redux-persist-filesystem-storage": "^1.3.2",
    "redux-persist-transform-encrypt": "^2.0.1",
    "redux-saga": "^0.16.0",
    "remotedev-serialize": "^0.1.8",
    "reselect": "^3.0.1",
    "rn-fetch-blob": "0.12.0",
    "rxjs": "^6.0.0",
    "rxjs-compat": "^6.0.0",
    "ts-optchain": "^0.1.3"
  },
  "devDependencies": {
    "@babel/cli": "^7.0.0",
    "@babel/core": "^7.8.4",
    "@babel/plugin-external-helpers": "^7.0.0",
    "@babel/plugin-proposal-optional-chaining": "^7.0.0",
    "@babel/preset-env": "^7.0.0",
    "@babel/preset-flow": "^7.12.1",
    "@babel/preset-typescript": "^7.7.4",
    "@babel/runtime": "^7.8.4",
    "@react-native-community/eslint-config": "^2.0.0",
    "@snowcoders/sortier": "2.6.2",
    "@types/lodash": "^4.14.121",
    "@types/moment": "^2.13.0",
    "@types/react": "^16.8.3",
    "@types/react-native": "^0.63.37",
    "@types/react-native-i18n": "^2.0.0",
    "@types/react-native-keep-awake": "^2.0.2",
    "@types/react-native-orientation": "^5.0.1",
    "@types/react-native-permissions": "^1.1.1",
    "@types/react-native-push-notification": "^5.0.5",
  
    "@types/react-redux": "^7.1.1",
    "@types/redux-form": "^8.1.4",
    "@types/redux-persist-transform-encrypt": "^2.0.1",
    "@typescript-eslint/eslint-plugin": "^4.9.1",
    "@typescript-eslint/parser": "^4.9.1",
    "babel-eslint": "^10.0.2",
    "babel-jest": "^25.1.0",
    "babel-plugin-lodash": "^3.3.4",
    "eslint": "^6.5.1",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-config-prettier": "^6.5.0",
    "eslint-import-resolver-typescript": "^2.3.0",
    "eslint-plugin-flowtype": "^2.50.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.1.1",
    "eslint-plugin-react": "^7.21.5",
    "eslint-plugin-react-hooks": "^2.1.2",
    "flow-bin": "^0.92.1",
    "husky": "^4.3.5",
    "jest": "^25.1.0",
    "jest-each": "^23.4.0",
    "jest-react-native": "^18.0.0",
    "jetifier": "1.6.2",
    "lint-staged": "^10.5.3",
    "metro-react-native-babel-preset": "^0.64.0",
    "patch-package": "^5.1.1",
    "prettier": "^1.19.1",
    "prettier-eslint": "^9.0.1",
    "react-native-codegen": "^0.0.6",
    "react-native-schemes-manager": "^2.0.0",
    "react-native-typescript-transformer": "^1.2.13",
    "react-test-renderer": "17.0.1",
    "typescript": "~3.7.2"
  },
  "resolutions": {
    "graceful-fs": "4.2.4"
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged",
      "post-merge": "git update-index --assume-unchanged ./ios/LaunchScreen.storyboard"
    }
  },
  "jest": {
    "collectCoverage": false,
    "collectCoverageFrom": [
      "**/*.{js,jsx}",
      "!**/node_modules/**",
      "!**/vendor/**",
      "!**/flow-typed/**",
      "!**/ios/**",
      "!**/android/**"
    ],
    "preset": "react-native",
    "setupFiles": [
      "<rootDir>/node_modules/appcenter/test/AppCenterMock.js",
      "<rootDir>/node_modules/appcenter-analytics/test/AppCenterAnalyticsMock.js",
      "<rootDir>/node_modules/appcenter-crashes/test/AppCenterCrashesMock.js",
      "<rootDir>/testenv.js"
    ],
    "transform": {
      "^.+\\.js$": "<rootDir>/node_modules/react-native/jest/preprocessor.js"
    }
  },
  "name": "charterapp",
  "private": true,
  "rnpm": {
    "assets": [
      "./src/assets/fonts",
      "./src/assets/images",
      "./src/assets/svg"
    ]
  },
  "scripts": {
    "adbrev": "adb reverse tcp:8081 tcp:8081 && adb reverse tcp:9090 tcp:9090",
    "android-device": "yarn run adbrev && yarn react-native run-android --variant=devDebug",
    "android-sim": "killall adb; yarn react-native run-android --variant=devDebug",
    "bundle_android_resources": "yarn react-native bundle --platform android --entry-file index.js --bundle-output android/app/src/main/assets/index.bundle --assets-dest  android/app/src/main/res/ --dev false",
    "lint": "eslint --fix ./src/**/*.{ts,js,tsx} && eslint --fix \"./App*.{ts,tsx,js}\"",
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "ios-device": "react-native run-ios --device",
    "prepare": "patch-package",
    "reset": "rm -rf ./node_modules && rm -rf yarn.lock && watchman watch-del-all && yarn",
    "start": "react-native start",
    "test": "jest",
    "test:watch": "jest --watch"
  },
  "version": "2.1.0"
}

```

#### `firebase.json` for react-native-firebase v6:

```json
{
  "react-native": {
    "messaging_android_notification_channel_id": "High-Priority"
  }
}

```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [x] my application is an AndroidX application?
- [x] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [x] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext {
        buildToolsVersion = "29.0.3"
        minSdkVersion = 21
        compileSdkVersion = 29
        targetSdkVersion = 29
        ndkVersion = "20.1.5948944"
        supportLibVersion = "28.0.0"
    }
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:4.1.0'
        classpath 'com.google.gms:google-services:4.2.0'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url("$rootDir/../node_modules/react-native/android")
        }
        maven {
            // Android JSC is installed from npm
            url("$rootDir/../node_modules/jsc-android/dist")
        }

        google()
        jcenter()
        maven { url 'https://www.jitpack.io' }
        maven {
            // Required for react-native-background-geolocation
            url("${project(':react-native-background-geolocation-android').projectDir}/libs")
        }
        maven {
            // react-native-background-fetch
            url("${project(':react-native-background-fetch').projectDir}/libs")
        }
    }
//    flatDir {
//        dirs 'libs'
//    }
//    maven {
//        url "$rootDir/../node_modules/react-native-background-fetch/android/libs"
//    }
//    mavenLocal()
}

project.ext.react = [
        nodeExecutableAndArgs: ["node", "--max_old_space_size=16384"]
]
ext {
    compileSdkVersion = 29
    targetSdkVersion = 29
    buildToolsVersion = "29.0.3"
    supportLibVersion = "28.0.0"
    playServicesVersion = "16.1.0"
}


subprojects { subproject ->
    afterEvaluate {
        if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                compileSdkVersion rootProject.ext.compileSdkVersion
                buildToolsVersion rootProject.ext.buildToolsVersion
            }
        }
    }
}

```

#### `android/app/build.gradle`:

```groovy
apply plugin: "com.android.application"

import com.android.build.OutputFile

/**
 * The preferred build flavor of JavaScriptCore.
 *
 * For example, to use the international variant, you can use:
 * `def jscFlavor = 'org.webkit:android-jsc-intl:+'`
 *
 * The international variant includes ICU i18n library and necessary data
 * allowing to use e.g. `Date.toLocaleString` and `String.localeCompare` that
 * give correct results when using with locales other than en-US.  Note that
 * this variant is about 6MiB larger per architecture than default.
 */
def jscFlavor = 'org.webkit:android-jsc:+'

project.ext.react = [
    entryFile: "index.js",
    bundleInDevDebug: true,
    bundleInDevRelease: true,
    bundleInStoreDebug: false,
    bundleInStoreRelease: true,
    bundleInBetaDebug: false,
    bundleInBetaRelease: true,
    enableHermes: false,  // clean and rebuild if changing
]

apply from: "../../node_modules/react-native/react.gradle"
//apply from: "../../node_modules/react-native-sentry/sentry.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

def enableHermes = project.ext.react.get("enableHermes", false);

/**
 * Purge Background Geolocation debug sounds from release build.
 */
def purgeBackgroundGeolocationDebugResources(applicationVariants) {
    applicationVariants.all { variant ->
        if (variant.buildType.name == 'release') {
            variant.mergeResources.doLast {
                delete(fileTree(dir: variant.mergeResources.outputDir, includes: ['raw_tslocationmanager*']))

            }
        }
    }
}

android {
    //Remove debug sounds from BackgroundGeolocation plugin
    purgeBackgroundGeolocationDebugResources(applicationVariants)
    ndkVersion rootProject.ext.ndkVersion
    packagingOptions {
        pickFirst 'lib/armeabi-v7a/libc++_shared.so'
        pickFirst 'lib/arm64-v8a/libc++_shared.so'
        pickFirst 'lib/x86_64/libc++_shared.so'
        pickFirst 'lib/x86/libc++_shared.so'
        pickFirst '**/libgnustl_shared.so'
        pickFirst '**/libjsc.so'
    }
    signingConfigs {
    }
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    compileSdkVersion rootProject.ext.compileSdkVersion

    defaultConfig {
        applicationId "com.bundle.id"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1585146055
        versionName '2.1.0'
        vectorDrawables.useSupportLibrary = true
        multiDexEnabled true
        missingDimensionStrategy 'react-native-camera', 'general'
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
    }
    buildTypes {
        release {
            //signingConfig signingConfigs.release
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
        debug {
            minifyEnabled enableProguardInReleaseBuilds
        }
    }
    flavorDimensions "default"
    productFlavors {
        beta {
            applicationId 'com.bundle.id.beta'
            resValue "string", "build_config_package", "com.bundle"
            manifestPlaceholders.geoLicense = "<redacted>"
        }
        dev {
            applicationId 'com.bundle.id.dev'
            resValue "string", "build_config_package", "com.bundle"
            manifestPlaceholders.geoLicense = "<redacted>"
        }
        store {
            applicationId 'com.bundle.id'
            resValue "string", "build_config_package", "com.bundle"
            manifestPlaceholders.geoLicense = "<redacted>"
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2, "arm64-v8a": 3, "x86_64": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                         defaultConfig.versionCode + versionCodes.get(abi)
            }
        }
    }
}

dependencies {
    implementation 'androidx.multidex:multidex:2.0.0'
    //noinspection GradleDynamicVersion
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation 'androidx.appcompat:appcompat:1.0.0'
    implementation 'com.facebook.react:react-native:+'
    implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
      exclude group:'com.facebook.fbjni'
    }
    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
        exclude group:'com.squareup.okhttp3', module:'okhttp'
    }
    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }
    if (enableHermes) {
        def hermesPath = "../../node_modules/hermes-engine/android/";
        debugImplementation files(hermesPath + "hermes-debug.aar")
        releaseImplementation files(hermesPath + "hermes-release.aar")
    } else {
        implementation jscFlavor
    }
    // Firebase dependencies
    // implementation "com.google.android.gms:play-services-base:16.1.0"
    implementation 'me.leolin:ShortcutBadger:1.1.21@aar' 
    implementation 'androidx.recyclerview:recyclerview:1.0.0'
}


// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
apply plugin: 'com.google.gms.google-services'

```

#### `android/settings.gradle`:
I have no settings.gradle. only properties.gradle, so added that
```groovy
# This option should only be used with decoupled projects. More details, visit
# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
# org.gradle.parallel=true
# AndroidX package structure to make it clearer which packages are bundled with the
# Android operating system, and which are packaged with your app's APK
# https://developer.android.com/topic/libraries/support-library/androidx-rn
android.useAndroidX=true
# Automatically convert third-party libraries to use AndroidX
android.enableJetifier=true
# Version of flipper SDK to use with React Native
FLIPPER_VERSION=0.54.0
org.gradle.daemon=true
org.gradle.parallel=true
org.gradle.configureondemand=true
org.gradle.jvmargs=-Xmx3g -XX:MaxPermSize=2048m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8


```

#### `MainApplication.java`:

```java
package com.bundle.id;

import android.content.Context;
import android.content.Intent;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.Bundle;

import androidx.multidex.MultiDexApplication;

import com.facebook.react.HeadlessJsTaskService;
import com.facebook.react.ReactApplication;
import com.facebook.react.ReactInstanceManager;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.soloader.SoLoader;
import com.facebook.react.PackageList;

import java.lang.reflect.InvocationTargetException;
import java.util.List;
import java.util.Arrays;

import com.reactnativecommunity.picker.RNCPickerPackage;

public class MainApplication extends MultiDexApplication implements ReactApplication {
    private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
        @Override
        public boolean getUseDeveloperSupport() {
            return BuildConfig.DEBUG;
        }

        @Override
        protected List<ReactPackage> getPackages() {
            @SuppressWarnings("UnnecessaryLocalVariable")
            List<ReactPackage> packages = new PackageList(this).getPackages();
            // Packages that cannot be autolinked yet can be added manually here, for example:
            // packages.add(new MyReactNativePackage());
            packages.add(new RNCPickerPackage());
            return packages;
        }

        @Override
        protected String getJSMainModuleName() {
            return "index";
        }
    };

    @Override
    public ReactNativeHost getReactNativeHost() {
        return mReactNativeHost;
    }

    public void onCreate() {
        super.onCreate();
        SoLoader.init(this, /* native exopackage */ false);
        initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
    }


    /**
     * Loads Flipper in React Native templates. Call this in the onCreate method with something like
     * initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
     *
     * @param context
     * @param reactInstanceManager
     */
    private static void initializeFlipper(
            Context context, ReactInstanceManager reactInstanceManager) {
        if (BuildConfig.DEBUG) {
            try {
        /*
         We use reflection here to pick up the class that initializes Flipper,
        since Flipper library is not available in release mode
        */
                Class<?> aClass = Class.forName("com.bundle.id.ReactNativeFlipper");
                aClass
                        .getMethod("initializeFlipper", Context.class, ReactInstanceManager.class)
                        .invoke(null, context, reactInstanceManager);
            } catch (ClassNotFoundException e) {
                e.printStackTrace();
            } catch (NoSuchMethodException e) {
                e.printStackTrace();
            } catch (IllegalAccessException e) {
                e.printStackTrace();
            } catch (InvocationTargetException e) {
                e.printStackTrace();
            }
        }
    }
}
```

#### `AndroidManifest.xml`:

```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    package="com.bundle.id">


    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />

    <application
        android:name=".MainApplication"
        android:allowBackup="false"
        android:icon="@mipmap/ic_launcher"
        android:label="@string/app_name"
        android:launchMode="singleTop"
        android:usesCleartextTraffic="true" 
        tools:targetApi="28"
        tools:ignore="GoogleAppIndexingWarning"
        android:networkSecurityConfig="@xml/network_security_config"
        android:theme="@style/SplashTheme">
        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
        <!-- react-native-background-geolocation licence -->
        <meta-data
            android:name="com.transistorsoft.locationmanager.license"
            android:value="${geoLicense}" />

        <activity
            android:name=".MainActivity"
            android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
            android:launchMode="singleTask"
            android:label="@string/app_name"
            android:windowSoftInputMode="adjustPan">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />

                <action android:name="android.intent.action.DOWNLOAD_COMPLETE" />
            </intent-filter>
        </activity>

        <service android:name="com.voximplant.foregroundservice.VIForegroundService" />
<!--        <meta-data-->
<!--            android:name="com.google.firebase.messaging.default_notification_color"-->
<!--            android:resource="@color/colorAccent" />-->
<!--        <meta-data-->
<!--            android:name="com.google.firebase.messaging.default_notification_channel_id"-->
<!--            android:value="@string/default_notification_channel_id" />-->
        
        <service android:name="com.bundle.id.MyTaskService" />
    </application>


</manifest>

```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
yarn react-native info
yarn run v1.22.10
$ /Users/roy/Documents/src/charter-app/node_modules/.bin/react-native info
info Fetching system and libraries information...
System:
    OS: macOS 11.1
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 57.79 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.3.0 - /var/folders/t5/hkglj_8s1lbb7tnf4jc4ywyh0000gn/T/yarn--1610458455841-0.3911375543839333/node
    Yarn: 1.22.10 - /var/folders/t5/hkglj_8s1lbb7tnf4jc4ywyh0000gn/T/yarn--1610458455841-0.3911375543839333/yarn
    npm: 7.0.14 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.2, DriverKit 20.0, macOS 11.0, tvOS 14.2, watchOS 7.1
    Android SDK:
      API Levels: 23, 25, 26, 27, 28, 29
      Build Tools: 27.0.3, 28.0.2, 28.0.3, 29.0.2, 29.0.3
      System Images: android-26 | Google APIs Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6953283
    Xcode: 12.2/12B45b - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_201 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.0-rc.1 => 0.64.0-rc.1
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [x] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `"@react-native-firebase/app": "10.4.0",`
- **`Firebase` module(s) you're using that has the issue:**
  - `"@react-native-firebase/messaging": "10.4.1",`
- **Are you using `TypeScript`?**
  - `Y` & `3.7.2`

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by mikehardy** [comment]
I believe you are responsible in your app for ensuring the notification channel exists, it sounds like it may not exist?

A full-fledged notification package should be used to do so, e.g. https://notifee.app/react-native/docs/android/channels ?

**Comment by royderksroute42** [comment]
Hi Mike,

The channel not existing before FireBase does its magic was my first thought aswel.

The react-native-push-notification package explicitly states in big bold letters that the code should NOT run within the React Context, thus it is defined outside anything React and in the "global scope"; which it is in our app.

Before logging in to our application, after which the firebase magic happens, I can already see the notification channel under App Info of the app in question as explained in detail on the Notifee document you have linked, and react-native-push-notification.CreateChannel callback has already logged that it was indeed created successfully.

For the sake of brevity I had a test that made sure that any FireBase calls are delayed by 10 seconds, and the creation of the notification channel was the first thing to run in the entry point for Javascript.
Unfortunately this did not prevent the error from showing up again.

Just a thought: `firebase().hasPermission()`, which comes before the call to `firebase().getToken()`, does not produce the error, may that narrow down the possibilities?

**Comment by mikehardy** [comment]
Very strange. That log indicates there might be an error / stack above it direct from the firebase-android-sdk, maybe that will have something of interest. The channel was more of a command error but it seems like you have that dialed in.

FIR_AUTH_ERROR is a different thing entirely normally I should have paid more attention. 

You're on fully up to date (even bleeding edge with node15/react-native 0.64rc) versions but cocoapods is old, obviously has no bearing here though :thinking: - have you overridden firebase-android-sdk anywhere?

You can remove all multidex items now that minSdkVersion is >= 21 (build.gradle and MultiDexApplication (also has no bearing though). Looks like some of your androidx deps are maybe out of date, I really don't think either matters here though. 

I can't see anything here that might cause it. All I can recommend is a single self-contained App.js that reproduces it along with a specific FCM JSON that - when posted to a device with the App.js running - will trigger it, then we can look more deeply

**Comment by royderksroute42** [comment]
Unfortunately, there is nothing other than that error in logcat, flipper,  react-native-debugger or yellowbox, so I'm kind of stuck at this point.

`have you overridden firebase-android-sdk anywhere`
I don't think i did, apart from some stackoverflow bandaids like the following:

```
subprojects { subproject ->
    afterEvaluate {
        if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                compileSdkVersion rootProject.ext.compileSdkVersion
                buildToolsVersion rootProject.ext.buildToolsVersion
            }
        }
    }
}
```

This whole situation was caused by the default Android notification channel not requesting popup permissions, which is why I resulted to defining my own channel and specifying that in the firebase.json.

Does there happen to be some configuration option i've missed for this library to request sound & popup (importance 4, if i understand correctly) on the default channel?

Thanks for the multidex tip, I'll follow through on those.

**Comment by mikehardy** [comment]
I don't know. I've *never* seen a crash that did not have a java stack trace in logcat. Unfiltered logcat may be important to watch in case you were filtering it. It is also possible to scan through the actual messaging code https://github.com/firebase/firebase-android-sdk/tree/master/firebase-messaging/src/main/java/com/google/firebase/messaging

**Comment by royderksroute42** [comment]
Hi Mike,

After checking three more times, I am confident that there is no error in the vicinity of the FIS_AUTH_ERROR; unfiltered logcat.

However, we've found a 'workaround' by doing:

```
firebase().getToken(...).catch(() => firebase().getToken()...) 
```

It's a long ways from ideal,  but the only way we can get this working, and we'll take it!
We haven't tried to isolate the error in a barebones app as we're fairly confident the error would not show in that situation, judging from experience. There must be some quirk as we have a fairly big codebase, even gating everything but the appregistry call itself until the `react-native-push-notification` confirms the channel is created did not help, so we've ruled out any race conditions as far as we're concerned.

Let's consider this one closed.
